Mathlib/Tactic/Linarith/Parsing.lean

```diff
@@ -94,9 +94,16 @@ def Sum.mul (s1 s2 : Sum) : Sum :=
     RBMap.empty
 
 /-- The `n`th power of `s : Sum` is the `n`-fold product of `s`, with `s.pow 0 = Sum.one`. -/
-def Sum.pow (s : Sum) : ℕ → Sum
-  | 0     => Sum.one
-  | (k+1) => s.mul (s.pow k)
+partial def Sum.pow (s : Sum) : ℕ → Sum
```

Comment:
I guess there's some overlap here with #8885. Perhaps worth leaving a comment about unifying things, though not worth waiting for that PR.

Reply:
No, that PR would not apply because it depends on `Sum` being lawful and having been proved as such.